docker/docker-tezos-packages.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2022 Oxhead Alpha
+# SPDX-License-Identifier: LicenseRef-MIT-OA
+
+import os
+import sys
+import shlex
+import subprocess
+from package.packages import packages
+from package.package_generator import parser
+from package.package_generator import output_dir as container_output_dir
+
+ubuntu_versions = [
+    "bionic",  # 18.04
+    "focal",  # 20.04
+    "jammy",  # 22.04
+]
+
+fedora_versions = [
+    "35",
+    "36",
+]
+
+
+def check_call(cmd):
+    return subprocess.check_call(shlex.split(cmd))
+
+
+def call(cmd):
+    return subprocess.call(shlex.split(cmd))
+
+
+def get_proc_output(cmd):
+    if sys.version_info.major == 3 and sys.version_info.minor < 7:
+        return subprocess.run(shlex.split(cmd), stdout=subprocess.PIPE)
+    else:
+        return subprocess.run(shlex.split(cmd), capture_output=True, text=True)
+
+
+def get_packages_to_build(arg):
+    if arg:
+        for pkg in arg:
+            if not packages.get(pkg, None):
+                raise Exception("Unknown package name.")
+        return {key: packages[key] for key in arg}
+    return packages
+
+
+if os.getenv("USE_PODMAN", None):
+    virtualisation_engine = "podman"
+else:
+    virtualisation_engine = "docker"
+
+parser.add_argument(
+    "--output-dir",
+    help="provide a directory to place the built packages",
+    default=f"{os.path.join(os.getcwd(), 'out')}",
+    type=os.path.abspath,
+)
+parser.add_argument(
+    "--build-sapling-package",
+    help="whether to build the sapling-params package",
+    action="store_true",
+)
+parser.set_defaults(build_sapling_package=False)
+args = parser.parse_args()
+
+docker_volumes = ""
+
+if args.binaries_dir:
+    binaries_dir_name = os.path.basename(args.binaries_dir)
+    docker_volumes = (
+        f"-v {args.binaries_dir}:/tezos-packaging/docker/{binaries_dir_name}"
+    )
+else:
+    binaries_dir_name = None
+
+target_os = args.os
+
+if target_os == "ubuntu":
+    if args.distributions:
+        for dist in args.distributions:
+            if dist not in ubuntu_versions:
+                raise Exception(
+                    f"Distribution {dist} is not supported for {target_os}."
+                )
+        distributions = args.distributions
+    else:
+        distributions = ubuntu_versions
+else:
+    if args.distributions:
+        for dist in args.distributions:
+            if dist not in fedora_versions:
+                raise Exception(
+                    f"Distribution {dist} is not supported for {target_os}."
+                )
+        distributions = args.distributions
+    else:
+        distributions = ["native"]
```

Comment:
Two suggestions here:
1. make a function for the part that's not in the outermost `if`/`else` and call it twice
2. replace the outermost `else` with an `elif` to check that the target os is `fedora` and fail otherwise